### ISSUE TITLE
GPU performance (Was named: Cannot load onnxruntime.capi)

### ISSUE THREAD

**Issue by Gaploid** [Microsoft/onnxruntime#625]
**Describe the bug**
`import onnxruntime as rt` faild with that error

C:\Users\gaplo\AppData\Local\Programs\Python\Python37\lib\site-packages\onnxruntime\capi\_pybind_state.py:12: UserWarning: Cannot load onnxruntime.capi. Error: 'DLL load failed: The specified module could not be found.'
  warnings.warn("Cannot load onnxruntime.capi. Error: '{0}'".format(str(e)))
Traceback (most recent call last):
  File "1.py", line 4, in <module>
    import onnxruntime as rt
  File "C:\Users\gaplo\AppData\Local\Programs\Python\Python37\lib\site-packages\onnxruntime\__init__.py", line 21, in <module>
    from onnxruntime.capi._pybind_state import RunOptions, SessionOptions, get_device, NodeArg, ModelMetadata
ImportError: cannot import name 'RunOptions' from 'onnxruntime.capi._pybind_state' (C:\Users\gaplo\AppData\Local\Programs\Python\Python37\lib\site-packages\onnxruntime\capi\_pybind_state.py)

**!!All Environment variables to CUDA are set!!!**

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- ONNX Runtime installed from (source or binary): pip install onnxruntime-gpu
- ONNX Runtime version: 0.2.1
- Python version: Python 3.7.2
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.1.105_win10
- GPU model and memory: NVIDIA GTX 1080

**To Reproduce**
Describe steps/code to reproduce the behavior:
`import onnxruntime as rt`

**Comment by Gaploid** [comment]
I`ve tried on Ubuntu 16.04, CUDA 10.1 and cuDNN 7.5 got the same error:
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-4-baa92e40428f> in <module>()
----> 1 import onnxruntime as rt

/anaconda/envs/py35/lib/python3.5/site-packages/onnxruntime/__init__.py in <module>()
     19 onnxruntime_validation.check_distro_info()
     20 from onnxruntime.capi.session import InferenceSession
---> 21 from onnxruntime.capi._pybind_state import RunOptions, SessionOptions, get_device, NodeArg, ModelMetadata

ImportError: cannot import name 'RunOptions'

**Comment by raymondxyang** [comment]
Just tested 0.2.1 and newly published onnxruntime-gpu 0.3.0 on ubuntu, import succeed. please make sure there is no broken develop build in your environment (check if there is cached path in easy-install.pth)

**Comment by Gaploid** [comment]
@raymondxyang  I did not use easy-install. It was pip install. Maybe I'm missing some packages or libs. I`ve installed Cuda, cuDNN and onnxruntime-gpu. Do I need anything else?

**Comment by raymondxyang** [comment]
it shouldnt need anything else.. could you go to where your package is installed to check if  capi/onnxruntime_pybind11_state.so presents there?

**Comment by Gaploid** [comment]
Yep, its here 
![image](https://user-images.githubusercontent.com/6171245/54395299-81fcbc00-46c0-11e9-8420-2c3bf740004a.png)

**Comment by raymondxyang** [comment]
ha thats weird. could you try delete the onnxruntime folder, then import, to see if you actually import from somewhere else?

**Comment by raymondxyang** [comment]
and onnxruntime_gpu-0.x.x.dist-info

**Comment by Gaploid** [comment]
```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-baa92e40428f> in <module>()
----> 1 import onnxruntime as rt

ImportError: No module named 'onnxruntime'

```

**Comment by raymondxyang** [comment]
I actually tested the onnxruntime-gpu.. could u try install onnxruntime-gpu==0.3.0 to see if the error still presents?

**Comment by Gaploid** [comment]
Yep, That's ver 0.3 
![image](https://user-images.githubusercontent.com/6171245/54396248-6515b800-46c3-11e9-9305-5b7e2f529e04.png)
Maybe I'm missing env paths or something like that?

**Comment by raymondxyang** [comment]
not sure on that tho.. just checked my environment list there's no other specific thing needs to point out.. I installed CUDA 9.1 and cuDNN 7.3 following https://docs.nvidia.com/cuda/archive/9.1/cuda-installation-guide-linux/index.html

**Comment by Gaploid** [comment]
hmm, I've installed Cuda 10.1 and cuDNN 7.5 can be that a problem on Ubuntu?

**Comment by raymondxyang** [comment]
yes. We need cuda 9.1 and cudnn 7.3 in order to run. (we are working on upgrading it)

**Comment by Gaploid** [comment]
Ahh !!! But what about Windows 10, it seems cuda 10 is supported there and I have the same error.

**Comment by raymondxyang** [comment]
We build the package with 9.1 as well on windows

**Comment by Gaploid** [comment]
Downgraded to Cuda 9.1 and it works. Difference CPU vs GPU(Tesla K80) around ~2 times faster.

**Comment by Gaploid** [comment]
Downgraded on windows to Cuda 9.1 and it works. But performance on GPU is much slower:
(GPU) -1.7 sec   vs (CPU) - 0.0199 sec.  89 times slower on GPU and Im clearly see that GPU is using in perf monitor. @raymondxyang  any ideas? GPU - Nvidia GTX 1080 with latest drivers on Windows 10 - 1809

**Comment by raymondxyang** [comment]
no idea on that.. I ll inform the team 😅

**Comment by raymondxyang** [comment]
@HectorSVC could u take a look on it? Thanks

**Comment by pranavsharma** [comment]
What model are you using? Please attach the model and test data so that we can debug.

**Comment by Gaploid** [comment]
Here is a link to onnx model and a few image examples. Onnx model was exported from customvision.ai
https://videoindex.blob.core.windows.net/html/model-data.zip 

Code example of invocation:
```
import glob
import os
import numpy as np
import onnxruntime as rt
import cv2
from datetime import datetime
import ntpath
import time


def main(arg):
	sess = rt.InferenceSession('apex11.onnx')   
	img1 = "out-up11736592.jpg"
	

	img = cv2.imread(img1)	

	ximg224 = cv2.resize(img, (224, 224))

	ximg = ximg224[None, :, :, :] #numpy.expand_dims(ximg224, axis=0)
	ximg = (ximg).astype(np.float32)

	ximg =np.rollaxis(ximg, 3, 1) 
	
	start = time.time()
	
	input_name = sess.get_inputs()[0].name
	res = sess.run(None, {input_name: ximg})
	prob = res[0]
	
	end = time.time()
	
	print(prob)
	print(end - start)

if __name__ == "__main__":
	import sys
	main(sys.argv[1])

```

**Comment by snnn** [comment]
> @pranavsharma Here is a link to onnx model and a few image examples. Onnx model was exported from customvision.ai
> https://videoindex.blob.core.windows.net/html/model-data.zip
> 
> Code example of invocation:
> 
> ```
> import glob
> import os
> import numpy as np
> import onnxruntime as rt
> import cv2
> from datetime import datetime
> import ntpath
> import time
> 
> 
> def main(arg):
> 	sess = rt.InferenceSession('apex11.onnx')   
> 	img1 = "out-up11736592.jpg"
> 	
> 
> 	img = cv2.imread(img1)	
> 
> 	ximg224 = cv2.resize(img, (224, 224))
> 
> 	ximg = ximg224[None, :, :, :] #numpy.expand_dims(ximg224, axis=0)
> 	ximg = (ximg).astype(np.float32)
> 
> 	ximg =np.rollaxis(ximg, 3, 1) 
> 	
> 	start = time.time()
> 	
> 	input_name = sess.get_inputs()[0].name
> 	res = sess.run(None, {input_name: ximg})
> 	prob = res[0]
> 	
> 	end = time.time()
> 	
> 	print(prob)
> 	print(end - start)
> 
> if __name__ == "__main__":
> 	import sys
> 	main(sys.argv[1])
> ```

You should do a few times warm up before measuring the performance.

**Comment by Gaploid** [comment]
@snnn Yep, I know, it helps a little. But the speed 80x difference is still there.

**Comment by HectorSVC** [comment]
I tried your model with our perf test tool, and use faked data with size [1, 3, 224, 224]. I saw gpu is 2x faster than cpu. I'm using Windows 10. 
CPU: Intel Xeon CPU E5- @ 2.00GHz (2 processors)
GPU: Nvidia Geforce GTX 1080

**Comment by faxu** [comment]
@Gaploid are you still seeing this perf gap? If so, can you provide more details on your exact test/machine config and input data so we can try to replicate for debugging?

**Comment by Gaploid** [comment]
here is my dxdiag info on achine with GTX 1080
![image](https://user-images.githubusercontent.com/6171245/55120582-b84a2a80-5107-11e9-865c-9edaa86071c4.png)

**Comment by virgile-thermo** [comment]
Hello, 
I have the same problem.
I tried : 
windows 10
cuda 10
cudnn for cuda 10, 7.5.0.56
onnxruntime-gpu 0.3.0

and 

windows 10
cuda 9.2
cudnn for cuda 9.2, 7.3.1.20
onnxruntime-gpu 0.3.0

and 

windows 10
cuda 9.1
cudnn for cuda 9.1, 7.1
onnxruntime-gpu 0.3.0

with cuda 9.1 the import onnxruntime as rt works!!!

but I don't know why?

Regards

**Comment by raymondxyang** [comment]
@virgile-thermo because our package was build against cuda 9.1. That's the library it will be looking for in your environment once you import it

**Comment by virgile-thermo** [comment]
obviously.
Thanks

**Comment by Rapternmn** [comment]
These issues are resolved by upgrading to onnxruntime 0.5.0
